src/tlapm_args.ml

```diff
@@ -134,140 +107,254 @@ let quote_if_needed s =
-    "--printfp", Arg.String print_fp,
-                 "<f> print the fingerprints stored in file <f> and quit";
-    "--usefp", Arg.String use_fp,
-               "<f> load fingerprints from file <f> (save as usual)";
-    "--fpp", Arg.Set fp_deb,
-             " print the fingerprints of obligations in toolbox messages";
-    "--cache-dir", Arg.String set_tlapm_cache_dir,
-            "<directory> save auxiliary and \
-             temporary files under <directory>. \
+open Cmdliner
+
+let version_t =
+  let doc = "Show version number and exit." in
+  Arg.(value & flag & info ["version"] ~doc)
+
+let verbose_t =
+  let doc = "Show version number and exit." in
+  Arg.(value & flag & info ["v"; "verbose"] ~doc)
+
+let where_t =
+  let doc = "Show location of standard library and exit." in
+  Arg.(value & flag & info ["where"] ~doc)
+
+let config_t =
+  let doc = "Show configuration and exit." in
+  Arg.(value & flag & info ["config"] ~doc)
+
+let summary_t =
+  let doc = "Show summary of theorems (implies -N and not -C)." in
+  Arg.(value & flag & info ["summary"] ~doc)
+
+let timing_t =
+  let doc = "Show runtime statistics." in
+  Arg.(value & flag & info ["timing"] ~doc)
+
+let search_dir_t =
+  let doc = "Add $(docv) to search path." in
+  Arg.(value & opt_all dir [] & info ["I"] ~docv:"DIR" ~doc)
+
+let keep_going_t =
+  let doc = "Keep going on backend failures." in
+  Arg.(value & flag & info ["k"] ~doc)
+
+let suppress_all_t =
+  let doc = "Do not run any backend verifiers." in
+  Arg.(value & flag & info ["N"] ~doc)
+
+let check_t =
+  let doc = "Check proofs in Isabelle/TLA+." in
+  Arg.(value & flag & info ["C"] ~doc)
+
+let threads_t =
+  let doc = "Set number of worker threads to $(docv)." in
+  Arg.(value & opt (some int) None & info ["threads"] ~docv:"N" ~doc)
+
+let method_t =
+  let doc = "Set default method to $(docv) (try --method=help)." in
+  Arg.(value & opt (some string) None & info ["method"] ~docv:"METH" ~doc)
+
+let solver_t =
+  let doc = "Set SMT solver to $(docv)." in
+  Arg.(value & opt (some string) None & info ["solver"] ~docv:"SOLVER" ~doc)
+
+let smt_logic_t =
+  let doc = "Set SMT logic to $(docv)." in
+  Arg.(value & opt (some string) None & info ["smt-logic"] ~docv:"LOGIC" ~doc)
+
+let fast_isabelle_t =
+  let doc = "(Windows-only) Launch Isabelle with fast shortcut." in
+  Arg.(value & flag & info ["fast-isabelle"] ~doc)
+
+let stretch_t =
+  let doc = "Multiply all timeouts by $(docv)." in
+  Arg.(value & opt float 1.0 & info ["stretch"] ~docv:"MULTIPLIER" ~doc)
+
+let noflatten_t =
+  let doc = "Do not flatten obligations." in
+  Arg.(value & flag & info ["noflatten"] ~doc)
+
+let nonormal_t =
+  let doc = "Do not normalize obligations before printing." in
+  Arg.(value & flag & info ["nonormal"] ~doc)
+
+let debug_t =
+  let doc = "Enable/disable debugging flags." in
+  Arg.(value & opt_all string [] & info ["debug"] ~docv:"[-]flag" ~doc)
+
+let toolbox_t =
+  let doc = "Toolbox mode." in
```

Comment:
The doc printer doesn't seem to handle `(some (pair int int))` and just prints out:
```
 --toolbox=VAL
           Toolbox mode.
```

Reply:
I will add a `~docv` for this option.